File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.java
class Solution {
    public int findMin(int[] nums) {

        int n = nums.length;
        int low = 0, high = n-1;
        int mid =- 1;

        while(low <= high){
            // if Sorted.
            mid = low+(high-low)/2;
            int ele = nums[mid];

            if(ele > nums[high]){
                low = mid+1;
            }
            else{
                high = high-1;
            }
        }

        return nums[mid];
        
    }
}

File: Data Structures & Algorithms/majority-element/submission-0.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:

        curr = nums[0]
        count = 1

        for ele in nums[1:]:
            if count == 0:
                curr = ele

            if ele == curr:
                count+=1
            else:
                count-=1

        return curr

        